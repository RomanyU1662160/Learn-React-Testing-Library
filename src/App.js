import React, { useState } from "react";
import logo from "./logo.svg";
import "./App.css";
import ToDoForm from "./components/ToDoForm";
import ToDoList from "./components/ToDoList";
import uuid from "uuid";

function App() {
  const init = [
    { id: 1, name: "Learn React.js", state: false },
    { id: 2, name: "Learn vue.js", state: false },
    { id: 3, name: "Learn Angular.js", state: false },
    { id: 4, name: "Learn Laravel", state: false }
  ];

  const [toDos, setToDos] = useState(init);

  const addToDo = val => {
    const newTodo = {
      id: uuid(),
      name: val,
      state: false
    };
    let newList = [...toDos, newTodo];
    setToDos(newList);
  };

  const DeleteTodo = id => {
    console.log("Delete clicked");
    const filteredToDos = toDos.filter(item => {
      return item.id !== id;
    });

    return setToDos(filteredToDos);
  };

  return (
    <div className="App">
      <h1 data-testid="list"> To do App </h1>
      <ToDoForm addToDo={addToDo}> </ToDoForm>
      <ToDoList list={toDos} deleteTodo={DeleteTodo}></ToDoList>
    </div>
  );
}

export default App;
