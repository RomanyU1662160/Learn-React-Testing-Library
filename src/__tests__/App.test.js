import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import App from "../App.js";

afterEach(cleanup);

it("matches snapshot", () => {
  const { asFragment } = render(<App></App>);
  expect(asFragment()).toMatchSnapshot();
});
