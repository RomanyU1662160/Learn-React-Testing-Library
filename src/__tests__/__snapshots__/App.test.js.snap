// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div>
      <form
        class="mt-4 col-md-6 offset-md-3"
      >
        <div
          class="form-group"
        >
          <input
            class="form-control"
            placeholder="Add to do"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <button
            class="btn-dark"
          >
            Add to do
          </button>
        </div>
      </form>
    </div>
    <div>
      <h3>
         to do list List
      </h3>
      <div
        class="row border-bottom p-4"
      >
        <div
          class="col-md-4"
        >
           Name : Learn React.js 
        </div>
        <div
          class="col-md-4"
        >
          Status:  not Yet 
        </div>
        <div
          class="col-md-2"
        >
          <button
            class="btn btn-danger"
          >
             Delete 
          </button>
        </div>
      </div>
      <div
        class="row border-bottom p-4"
      >
        <div
          class="col-md-4"
        >
           Name : Learn vue.js 
        </div>
        <div
          class="col-md-4"
        >
          Status:  not Yet 
        </div>
        <div
          class="col-md-2"
        >
          <button
            class="btn btn-danger"
          >
             Delete 
          </button>
        </div>
      </div>
      <div
        class="row border-bottom p-4"
      >
        <div
          class="col-md-4"
        >
           Name : Learn Angular.js 
        </div>
        <div
          class="col-md-4"
        >
          Status:  not Yet 
        </div>
        <div
          class="col-md-2"
        >
          <button
            class="btn btn-danger"
          >
             Delete 
          </button>
        </div>
      </div>
      <div
        class="row border-bottom p-4"
      >
        <div
          class="col-md-4"
        >
           Name : Learn Laravel 
        </div>
        <div
          class="col-md-4"
        >
          Status:  not Yet 
        </div>
        <div
          class="col-md-2"
        >
          <button
            class="btn btn-danger"
          >
             Delete 
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
